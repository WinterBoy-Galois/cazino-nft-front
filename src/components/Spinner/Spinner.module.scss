@import '../../styles/shared';

@keyframes circle-rotation {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(359deg);
  }
}

.spinner {
  width: 100%;
  height: 100%;
  clear: both;
  display: inline-block;
  vertical-align: text-bottom;
  border: 0.2em rgba($white, 0.25) solid;
  border-top: 0.2em rgba($white, 0.75) solid;
  border-radius: 50%;
  -webkit-animation: circle-rotation 0.6s infinite linear;
  animation: circle-rotation 0.6s infinite linear;

  &--white {
    border: 0.2em rgba($white, 0.25) solid;
    border-top: 0.2em rgba($white, 0.75) solid;
  }

  &--primary {
    border: 0.2em rgba($light-blue, 0.25) solid;
    border-top: 0.2em rgba($light-blue, 0.75) solid;
  }
}
