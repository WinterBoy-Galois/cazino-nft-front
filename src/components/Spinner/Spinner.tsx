import React from 'react';
import styles from './Spinner.module.scss';

interface IProps {
  color: 'WHITE' | 'PRIMARY';
  className?: string;
}

const Spinner: React.FC<IProps> = ({ color, className = '' }) => {
  const getClassName = () => {
    switch (color) {
      case 'PRIMARY':
        return `${styles.spinner} ${styles['spinner--primary']} ${className}`;
      case 'WHITE':
      default:
        return `${styles.spinner} ${styles['spinner--white']} ${className}`;
    }
  };

  return <span className={getClassName()} />;
};

export default Spinner;
