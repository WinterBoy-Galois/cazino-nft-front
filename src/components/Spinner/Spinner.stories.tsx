import React from 'react';
import { storiesOf } from '@storybook/react';
import Spinner from '.';

storiesOf('Components/Spinner', module)
  .add('White', () => (
    <div style={{ width: '100px', height: '100px' }}>
      <Spinner color="WHITE" />
    </div>
  ))
  .add('Primary', () => (
    <div style={{ width: '100px', height: '100px' }}>
      <Spinner color="PRIMARY" />
    </div>
  ));
