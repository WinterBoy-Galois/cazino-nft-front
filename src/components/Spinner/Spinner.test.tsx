import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Spinner from '.';

describe('<Spinner />', () => {
  beforeEach(cleanup);

  it('renders correctly', () => {
    const { container } = render(<Spinner color="WHITE" />);
    expect(container).toMatchSnapshot();
  });
});
