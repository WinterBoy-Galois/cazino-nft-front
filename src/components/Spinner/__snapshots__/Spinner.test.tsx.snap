// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders correctly 1`] = `
<div>
  <span
    class="spinner spinner--white "
  />
</div>
`;
